contracts/synthVault.sol

```diff
@@ -0,0 +1,212 @@
+    event MemberDeposits(address indexed token, address indexed member, uint newDeposit, uint totalDeposit, uint weight, uint totalWeight);
+    event MemberWithdraws(address indexed token, address indexed member, uint amount, uint weight, uint totalWeight);
+    event MemberHarvests(address indexed token, address indexed member, uint amount);
+
+   function setParams(uint one, uint two, uint three, uint four) external onlyDAO {
+        erasToEarn = one;
+        minimumDepositTime = two;
+        blockDelay = three;
+        vaultClaim = four;
+    }
+
+   //======================================DEPOSITS========================================//
+
+    // Holders to deposit for Interest Payments
+    function deposit(address token, uint amount) external {
+        depositForMember(token, msg.sender, amount);
+    }
+    
+    function depositForMember(address token, address member, uint amount) public {
+        require(iSYNTHFACTORY(_DAO().SYNTHFACTORY()).isSynth(token), "!synth");
+         getFunds(token, amount);
+        _deposit(token, member, amount);
+    }
+    function _deposit(address _token, address _member, uint _amount) internal {
+        mapMemberToken_lastTime[_member][_token] = block.timestamp;
+        mapMemberToken_deposit[_member][_token] += _amount; // Record balance for member
+        uint _weight = iUTILS(_DAO().UTILS()).calcSpotValueInBase(iSYNTH(_token).LayerONE(), _amount);
+        mapToken_totalFunds[_token] += _amount;
+        mapMember_weight[_member] += _weight;
+        totalWeight += _weight;
+        emit MemberDeposits(_token, _member, _amount, mapToken_totalFunds[_token], _weight, totalWeight);
+    }
+
+     //====================================== HARVEST ========================================//
+
+    function harvest(address token) external returns(uint reward) {
+        address _member = msg.sender;
+        reward = calcCurrentReward(token, _member);
+        mapMemberToken_lastTime[_member][token] = block.timestamp;
+        mapMemberToken_reward[_member][token] += reward;
+        totalRewards += reward;
+        emit MemberHarvests(token, _member, reward);
+        return reward;
+    }
+
+    function calcCurrentReward(address token, address member) public view returns(uint reward) {
+        uint _secondsSinceClaim = block.timestamp - mapMemberToken_lastTime[member][token];        // Get time since last claim
+        uint _share = calcReward(member);                                        
+        reward = (_share * _secondsSinceClaim) / iBASE(BASE).secondsPerEra();  
+        return reward;
+    }
+
+    function calcReward(address member) public view returns(uint) {
+        uint _weight = mapMember_weight[member];
+        uint _reserve = reserveBASE() / erasToEarn;  
+        uint _vaultReward = (_reserve * vaultClaim) / 10000;                   
+        return iUTILS(_DAO().UTILS()).calcShare(_weight, totalWeight, _vaultReward);         // Get member's share of that
+     }
```

Comment:
Functions to calc synth rewards and harvest.
Harvest rewards are not realised until the user withdraws from the synthsVault